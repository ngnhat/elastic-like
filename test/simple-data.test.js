/**
 * Created by ngnhat on Sat May 25 2019
 */
const Store = require('../index');
const simpleData = require('./example-data/simple-data.json');

const store = new Store();

beforeAll(() => {
  simpleData.forEach((data) => {
    store.add(data);
  });
});

describe('simple data', () => {
  it('keyword: cao su', () => {
    expect(store.search('cao su')).toMatchSnapshot();
  });

  it('keyword: sp000740', () => {
    expect(store.search('sp000740')).toMatchSnapshot();
  });
});
