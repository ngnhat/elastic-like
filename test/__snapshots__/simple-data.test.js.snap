// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple data keyword: cao su 1`] = `
Array [
  Object {
    "score": 8.262507460231904,
    "source": Object {
      "AllowsSale": true,
      "AttributeLabel": null,
      "Code": "SP000740",
      "FullName": "cao su ngón chân",
      "Id": 1791465,
      "IsLotSerialControl": null,
      "MasterCode": "SPC1903080932116664",
      "MasterUnitId": null,
      "ModifiedDate": "2019-04-26T09:11:26.8970000+07:00",
      "Name": "cao su ngón chân",
      "OnHand": -78,
      "ProductSerials": Array [],
      "isActive": true,
      "isDeleted": null,
    },
  },
  Object {
    "score": 8.262507460231904,
    "source": Object {
      "AllowsSale": true,
      "AttributeLabel": "nude",
      "Code": "SP000738",
      "FullName": "cao su gót chân - nude",
      "Id": 1791466,
      "IsLotSerialControl": null,
      "MasterCode": "SPC1901181539268125",
      "MasterUnitId": null,
      "ModifiedDate": "2019-04-26T09:27:43.4770000+07:00",
      "Name": "cao su gót chân",
      "OnHand": -9824,
      "ProductSerials": Array [],
      "isActive": true,
      "isDeleted": null,
    },
  },
  Object {
    "score": 8.262507460231904,
    "source": Object {
      "AllowsSale": true,
      "AttributeLabel": "nude",
      "Code": "SP001742",
      "FullName": "cao su gót chân - nude",
      "Id": 2440079,
      "IsLotSerialControl": null,
      "MasterCode": "SPC1901181539268125",
      "MasterUnitId": null,
      "ModifiedDate": "2019-05-15T14:43:29.6170000+07:00",
      "Name": "cao su gót chân",
      "OnHand": null,
      "ProductSerials": Array [],
      "isActive": true,
      "isDeleted": null,
    },
  },
  Object {
    "score": 8.262507460231904,
    "source": Object {
      "AllowsSale": true,
      "AttributeLabel": "nude",
      "Code": "SP001743",
      "FullName": "cao su gót chân - nude",
      "Id": 2440080,
      "IsLotSerialControl": null,
      "MasterCode": "SPC1901181539268125",
      "MasterUnitId": null,
      "ModifiedDate": "2019-05-15T14:43:58.1000000+07:00",
      "Name": "cao su gót chân",
      "OnHand": null,
      "ProductSerials": Array [],
      "isActive": true,
      "isDeleted": null,
    },
  },
  Object {
    "score": 8.262507460231904,
    "source": Object {
      "AllowsSale": true,
      "AttributeLabel": "trắng",
      "Code": "SP000739",
      "FullName": "cao su gót chân - trắng",
      "Id": 1791711,
      "IsLotSerialControl": null,
      "MasterCode": "SPC1901181539268125",
      "MasterUnitId": null,
      "ModifiedDate": "2019-03-08T09:32:12.1030000+07:00",
      "Name": "cao su gót chân",
      "OnHand": 189,
      "ProductSerials": Array [],
      "isActive": true,
      "isDeleted": null,
    },
  },
  Object {
    "score": 7.414827421233164,
    "source": Object {
      "AllowsSale": true,
      "AttributeLabel": null,
      "Code": "HH000009",
      "FullName": "Kẹo cao su tổng hợp",
      "Id": 2402636,
      "IsLotSerialControl": null,
      "MasterCode": "SPC190515103033574",
      "MasterUnitId": null,
      "ModifiedDate": "2019-05-15T10:30:33.0870000+07:00",
      "Name": "Kẹo cao su tổng hợp",
      "OnHand": 5,
      "ProductSerials": Array [],
      "isActive": true,
      "isDeleted": null,
    },
  },
]
`;

exports[`simple data keyword: sp000740 1`] = `
Array [
  Object {
    "score": 7.738021433760902,
    "source": Object {
      "AllowsSale": true,
      "AttributeLabel": null,
      "Code": "SP000740",
      "FullName": "cao su ngón chân",
      "Id": 1791465,
      "IsLotSerialControl": null,
      "MasterCode": "SPC1903080932116664",
      "MasterUnitId": null,
      "ModifiedDate": "2019-04-26T09:11:26.8970000+07:00",
      "Name": "cao su ngón chân",
      "OnHand": -78,
      "ProductSerials": Array [],
      "isActive": true,
      "isDeleted": null,
    },
  },
]
`;
